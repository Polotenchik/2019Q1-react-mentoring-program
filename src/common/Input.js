
import React from 'react';
import PropTypes from 'prop-types';

export const Input = () => {
    return (
        <input 
            className={'search-input'}
            type='text'
            placeholder = 'Start typing...'
        />
    );
};

