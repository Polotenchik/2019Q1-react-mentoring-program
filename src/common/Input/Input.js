import React from 'react';
import PropTypes from 'prop-types';

const Input = ({
  onStartTyping, onKeyEnterPressed, placeholder, value,
}) => (
    <input
        className={ 'search-input' }
        type='text'
        placeholder={ placeholder }
        onInput = { onStartTyping }
        onKeyPress={ onKeyEnterPressed }
        value = { value }
    />
);

Input.propTypes = {
  placeholder: PropTypes.string,
  onStartTyping: PropTypes.func.isRequired,
  onKeyEnterPressed: PropTypes.func,
  value: PropTypes.string,
};

Input.defaultProps = {
  placeholder: 'Start typing...',
  value: '',
};

export default Input;
