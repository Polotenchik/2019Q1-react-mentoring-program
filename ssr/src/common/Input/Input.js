import React from 'react';
import PropTypes from 'prop-types';

export const Input = ({ onStartTyping,  onKeyEnterPressed, placeholder, value }) => {
    return (
        <input 
            className={ 'search-input' }
            type='text'
            placeholder={ placeholder }
            onInput = { onStartTyping }
            onKeyPress={ onKeyEnterPressed }
            value = { value } 
        />
    );
};

Input.propTypes = {
    placeholder: PropTypes.string,
};
  
Input.defaultProps = {
    placeholder: 'Start typing...',
    value: ''
};