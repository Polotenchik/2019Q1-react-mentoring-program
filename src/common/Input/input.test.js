import React from 'react';
import { Input } from './Input';
import { shallow } from 'enzyme';

const mockText = 'foo bar quux';

describe('<Input />', () => {
  describe('Rendering', () => {
    it('should render input component', () => {
      const wrapper = shallow(<Input
        placeholder={ mockText }
      />);
      expect(wrapper).toMatchSnapshot();
    });

    it('should render input component', () => {
      const wrapper = shallow(<Input />);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
