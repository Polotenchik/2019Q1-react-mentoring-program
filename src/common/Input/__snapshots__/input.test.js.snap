// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> Rendering should render input component 1`] = `
<input
  className="search-input"
  placeholder="foo bar quux"
  type="text"
/>
`;

exports[`<Input /> Rendering should render input component 2`] = `
<input
  className="search-input"
  placeholder="Start typing..."
  type="text"
/>
`;
